File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int ans = INT_MIN;
        int l = 0, r = n-1;
        while(l<r){
            int h = min(height[l], height[r]);
            int w = r-l;
            ans = max(ans, h*w);
            height[l] < height[r] ? l++ : r--;
        }
        return ans;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i, j = 0, (len(height)-1)
       
        ans = float('-inf')
        while(i<j):
            area = (j-i)*min(height[i], height[j])
            ans = max(ans, area)
            if height[i] < height[j]:
                i+=1
            else:
                j-=1
        
        return ans